### PR TITLE
[Gleam]: first attempt to add Gleam.

### PR DESCRIPTION
**Pull request by whiplash**
Initial work for https://github.com/the-benchmarker/web-frameworks/issues/7386

### CODE REVIEW

File: gleam/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{{#language.image}}
+  FROM gleamlang/gleam
```
Comment (waghanza) [inline gleam/Dockerfile:2]: is there a maintained version of a docker image ?

File: gleam/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{{/language.image}}
+{{^language.image}}
+  FROM
```
Comment (waghanza) [inline gleam/Dockerfile:5]: ???

File: gleam/tmist/config.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+framework:
+  website: https://github.com/rawhat/mist
+  version: "3.0.0"
```
Comment (waghanza) [inline gleam/tmist/config.yaml:3]:
```suggestion
  version: 3.0
```

File: gleam/tmist/config.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+framework:
+  website: https://github.com/rawhat/mist
```
Comment (waghanza) [inline gleam/tmist/config.yaml:2]:
```suggestion
  github: rawhat/mist
```

File: gleam/Dockerfile (hunk 1 above)
Comment (waghanza) [reply gleam/Dockerfile:2]: According to https://github.com/gleam-lang/gleam/pkgs/container/gleam it should be 
```suggestion
  FROM ghcr.io/gleam-lang/gleam:v1.6.3-elixir
```

File: gleam/tmist/gleam.toml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# licences = ["Apache-2.0"]
+# repository = { type = "github", user = "", repo = "" }
+# links = [{ title = "Website", href = "" }]
+#
+# For a full reference of all the available options, you can have a look at
+# https://gleam.run/writing-gleam/gleam-toml/.
+
+[dependencies]
+gleam_stdlib = ">= 0.34.0 and < 2.0.0"
+mist = ">= 3.0.0 and < 4.0.0"
```
Comment (waghanza) [inline gleam/tmist/gleam.toml:17]:
```suggestion
mist = ">= 3.0.0 and < 3.1.0"
```

File: gleam/Dockerfile (hunk 1 above)
Comment (waghanza) [reply gleam/Dockerfile:2]: or maybe 
```dockerfile
FROM ghcr.io/gleam-lang/gleam:v1.6.3-erlang
```

File: gleam/tmist/gleam.toml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+gleam_erlang = ">= 0.32.0 and < 1.0.0"
```
Comment (waghanza) [inline gleam/tmist/gleam.toml:18]: not needed -> https://github.com/rawhat/mist/blob/d00d66e1e28bc72e8a47a6267684cced0411e180/gleam.toml#L15

File: gleam/tmist/gleam.toml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+gleam_http = ">= 3.7.1 and < 4.0.0"
```
Comment (waghanza) [inline gleam/tmist/gleam.toml:19]: not needed -> https://github.com/rawhat/mist/blob/d00d66e1e28bc72e8a47a6267684cced0411e180/gleam.toml#L16

File: gleam/tmist/gleam.toml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+gleam_otp = ">= 0.14.1 and < 1.0.0"
```
Comment (waghanza) [inline gleam/tmist/gleam.toml:20]: not needed -> https://github.com/rawhat/mist/blob/d00d66e1e28bc72e8a47a6267684cced0411e180/gleam.toml#L17

File: gleam/tmist/gleam.toml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+[dev-dependencies]
+gleeunit = ">= 1.0.0 and < 2.0.0"
```
Comment (waghanza) [inline gleam/tmist/gleam.toml:23]: needed ?

File: gleam/tmist/manifest.toml
Comment (waghanza) [inline gleam/tmist/manifest.toml:1]: could not this be generated onto containers ? I mean using `bootstrap` config option

**Review by waghanza — CHANGES_REQUESTED** [review]

File: gleam/tmist/gleam.toml (hunk 9 above)
Comment (whiplash) [reply gleam/tmist/gleam.toml:23]: maybe for testing

File: gleam/Dockerfile (hunk 1 above)
Comment (whiplash) [reply gleam/Dockerfile:2]: I wish somebody using gleam would give us hints

**Pull request merged**